enhancements/20251015-support-ipconfig-on-clusternetworks.md

```diff
@@ -0,0 +1,416 @@
+# Title
+
+This feature supports creation of vlan sub interfaces on mgmt and user defined cluster networks and configuration of IP addresses on these sub interfaces.
+
+## Summary
+
+This feature allows multiple virtual interfaces to be created per vlan on all cluster networks which can be used for L3 connectivity from Harvester Nodes to the external environment.
+
+### Related Issues
+
+https://github.com/harvester/harvester/issues/8101
+
+## Motivation
+
+- Some data‑plane or storage networks must be reachable as routed L3 subnets (for example: dedicated storage network with static or dynamic IP addressing)
+
+- Operators may want to isolate management traffic from application traffic using different routed subnets on separate physical NICs (different gateways, different next‑hop routers, or different physical uplinks).
+
+- Cloud/edge environments sometimes require that a non‑management NIC has its own IP address on a different L3 network to integrate with existing routing/OSPF/BGP or external services.
+
+Harvester currently configures a single VLAN sub-interface on the management NIC by default during installation. However, users need the ability to configure multiple VLAN sub-interfaces on both the management NIC and other user defined cluster networks with full L3 support.These configurations must be automated and persist across reboots and upgrades. Manual setup today is error-prone and not sustainable for production use.
+
+### Goals
+
+- Support creation of vlan sub interfaces on Harvester hosts when users configures a VM Network with L3Connectivy requires as `true`
+- Support assignment of IP addresses to the vlan sub interfaces on Harvester hosts by static method or dynamically using DHCP.
+
+### Non-goals
+
+- Supporting L3 Connectivity on VM Network with type `L2UntaggedNetwork`,`OverlayNetwork` and `L2VlanTrunkNetwork`.
+- Supporting IPPools or external IPAM integration for IP allocation to the vlan sub interfaces on Harvester hosts.
+- Support IPv6 addresses on the vlan sub interfaces of Harvester Hosts.
+
+## Proposal
+
+- Allow users to configure/unconfigure Layer 3 network Information on the VM Network resource.
+- Harvester network controller NAD handles the change and updates the annotations of the cluster network resource.
+- Harvester network controller cluster network agent handles the bridge vlan and creation/update/deletion of nmconnection profiles on the host.
```

Comment:
What's the reason behind the decision to let NetworkManager, but not the network controller, manage bridge VLANs and sub-interfaces? We used to do this ourselves entirely only except for the `mgmt` network.